src/app/Diagnostics/BulkEditHeapDumpLabels.tsx

```diff
@@ -59,28 +60,34 @@ export const BulkEditHeapDumpLabels: React.FC<BulkEditLabelsProps> = ({ checkedI
     setLoading(true);
     const tasks: Observable<unknown>[] = [];
     const toDelete = savedCommonLabels.filter((label) => !includesLabel(commonLabels, label));
-
-    heapDumps.forEach((r: HeapDump) => {
-      const idx = getIdxFromHeapDump(r);
-      if (checkedIndices.includes(idx)) {
-        const updatedLabels = [...r.metadata.labels, ...commonLabels].filter(
-          (label) => !includesLabel(toDelete, label),
-        );
-
-        tasks.push(context.api.postHeapDumpMetadata(r.heapDumpId, updatedLabels).pipe(first()));
-      }
-    });
     addSubscription(
-      forkJoin(tasks).subscribe({
-        next: handlePostUpdate,
-        error: handlePostUpdate,
+      propsTarget.subscribe((t) => {
+        if (!t) {
+          return;
```

Comment:
Minor nitpick, I think this would be more idiomatic:

```typescript
propsTarget
  .pipe(filter(t => !!t))
  .subscribe((t) => { // t is truthy here
```